libnetdata/onewayalloc/onewayalloc.c

```diff
@@ -36,7 +38,7 @@ static OWA_PAGE *onewayalloc_create_internal(OWA_PAGE *head, size_t size_hint) {
 
     // make sure the new page will fit both the requested size
     // and the OWA_PAGE structure at its beginning
-    size_hint += sizeof(OWA_PAGE);
+    size_hint += natural_alignment(sizeof(OWA_PAGE));
```

Comment:
This will still be unaligned if `(size_hint % OWA_NATURAL_ALIGNMENT) != 0`, eg. `size_hint = 2 * OWA_NATURAL_PAGE_SIZE + 1`. I'm not sure we care for such cases though.

Reply:
size_hint is aligned to a page size just below. Since we may need to pad the first allocation, the one just after `OWA_PAGE`, we needed to do this here. Otherwise, we were risking to overflow the page we will allocate.

Keep in mind that calls to `onewayalloc_create_internal()` are only made to request a new page, which happens through `onewayalloc_mallocz()` which calls `natural_alignment(size)` before requesting a new page. So, for all allocations made by the callers, the given `size_hint` to this function is aligned too.